google/resource_compute_instance_migrate.go

```diff
@@ -489,3 +496,17 @@ func getDiskFromAutoDeleteAndImage(config *Config, instance *compute.Instance, a
 
 	return nil, fmt.Errorf("could not find attached disk with image %q", image)
 }
+
+func migrateStateV5toV6(is *terraform.InstanceState) (*terraform.InstanceState, error) {
+	log.Printf("[DEBUG] Attributes before migration: %#v", is.Attributes)
+	if is.Attributes["boot_disk.0.initialize_params.#"] == "1" {
+		if (is.Attributes["boot_disk.0.initialize_params.0.size"] == "0" ||
+			is.Attributes["boot_disk.0.initialize_params.0.size"] == "") &&
+			is.Attributes["boot_disk.0.initialize_params.0.type"] == "" &&
+			is.Attributes["boot_disk.0.initialize_params.0.image"] == "" {
+			is.Attributes["boot_disk.0.initialize_params.#"] = "0"
```

Comment:
Is it worth calling `delete(is.Attributes, "boot_disk.0.initialize_params.INSERT_EVERY_ATTRIBUTE_HERE")` to get a properly clean state?

Reply:
Sure, can't hurt. Done.